test/e2e-go/features/incentives/whalejoin_test.go

```diff
@@ -0,0 +1,324 @@
+// online without immediate suspension.  This tests for a bug we had where we
+// calcululated expected proposal interval using the _old_ totals, rather than
+// the totals following the keyreg. So big joiner was being expected to propose
+// in the same block it joined.
+func TestWhaleJoin(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	defer fixtures.ShutdownSynchronizedTest(t)
+
+	t.Parallel()
+	a := require.New(fixtures.SynchronizedTest(t))
+
+	var fixture fixtures.RestClientFixture
+	// Make rounds shorter and seed lookback smaller, otherwise we need to wait
+	// 320 slow rounds for particpation effects to matter.
+	const lookback = 32
+	fixture.FasterConsensus(protocol.ConsensusFuture, time.Second, lookback)
+	fixture.Setup(t, filepath.Join("nettemplates", "Payouts.json"))
+	defer fixture.Shutdown()
+
+	// Overview of this test:
+	// 1. Take wallet15 offline (but retain keys so can back online later)
+	// 2. Have wallet01 spend almost all their algos
+	// 3. Wait for balances to flow through "lookback"
+	// 4. Rejoin wallet15 which will have way more stake that what is online.
+
+	clientAndAccount := func(name string) (libgoal.Client, model.Account) {
+		c := fixture.GetLibGoalClientForNamedNode(name)
+		accounts, err := fixture.GetNodeWalletsSortedByBalance(c)
+		a.NoError(err)
+		a.Len(accounts, 1)
+		fmt.Printf("Client %s is %v\n", name, accounts[0].Address)
+		return c, accounts[0]
+	}
+
+	c15, account15 := clientAndAccount("Node15")
+	c01, account01 := clientAndAccount("Node01")
+
+	// 1. take wallet15 offline
+	keys := offline(&fixture, a, c15, account15.Address)
+
+	// 2. c01 starts with 100M, so burn 99.9M to get total online stake down
+	burn, err := c01.SendPaymentFromUnencryptedWallet(account01.Address, basics.Address{}.String(),
+		1000, 99_900_000_000_000, nil)
+	a.NoError(err)
+	receipt, err := fixture.WaitForConfirmedTxn(uint64(burn.LastValid), burn.ID().String())
+	a.NoError(err)
+
+	// 3. Wait lookback rounds
+	_, err = c01.WaitForRound(*receipt.ConfirmedRound + lookback)
+	a.NoError(err)
+
+	// 4. rejoin, with 1.5B against the paltry 100k that's currently online
+	online(&fixture, a, c15, account15.Address, keys)
+
+	// 5. wait for agreement balances to kick in (another lookback's worth, plus some slack)
+	_, err = c01.WaitForRound(*receipt.ConfirmedRound + 2*lookback + 5)
+	a.NoError(err)
+
+	data, err := c15.AccountData(account15.Address)
+	a.NoError(err)
+	a.Equal(basics.Online, data.Status)
+
+	// even after being in the block to "get noticed"
+	txn, err := c15.SendPaymentFromUnencryptedWallet(account15.Address, basics.Address{}.String(),
+		1000, 1, nil)
+	a.NoError(err)
+	_, err = fixture.WaitForConfirmedTxn(uint64(txn.LastValid), txn.ID().String())
+	a.NoError(err)
+	data, err = c15.AccountData(account15.Address)
+	a.NoError(err)
+	a.Equal(basics.Online, data.Status)
+}
+
+// TestBigJoin shows that even though an account can't vote during the first 320
```

Comment:
I like this test: the intent should be that one cannot be challenged for any reason based on not voting during the 320 right?

Reply:
It doesn't actually _show_ that's true of challenges, but absenteeism and challenges are both "protected" by the free heartbeat that makes this work.